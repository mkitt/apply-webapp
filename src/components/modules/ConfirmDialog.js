// @flow
import React from 'react'
import Dialog from '../blocks/Dialog'

type Props = {
  handleCancel: Function,
  handleConfirm: Function,
}

export default({ handleCancel, handleConfirm }: Props) => (
  <Dialog
    confirmLabel="Yes, I'm sure"
    handleCancel={handleCancel}
    handleConfirm={handleConfirm}
    title="Are you sure?"
  >
    Clicking yes will confirm the action. Clicking cancel will abort this dialog.
  </Dialog>
)
