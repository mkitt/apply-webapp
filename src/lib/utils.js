// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
