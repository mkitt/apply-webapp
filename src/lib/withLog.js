import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
