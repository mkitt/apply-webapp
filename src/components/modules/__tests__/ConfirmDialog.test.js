import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from '../../../lib/utils'
import ConfirmDialog from '../ConfirmDialog'

describe('<ConfirmDialog />', () => {
  it('shows the correct snapshot tree for the <ConfirmDialog> module', () => {
    const component = (
      <ConfirmDialog
        handleCancel={noop}
        handleConfirm={noop}
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
