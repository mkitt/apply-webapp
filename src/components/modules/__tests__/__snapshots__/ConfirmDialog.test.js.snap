// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialog /> shows the correct snapshot tree for the <ConfirmDialog> module 1`] = `
.glamor-5,
[data-glamor-5] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 112px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 34px;
  font-weight: 200;
  color: #808080;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-1,
[data-glamor-1] {
  width: 100%;
  margin-top: 16px;
  margin-bottom: 32px;
  font-size: 14px;
  color: #808080;
}

.glamor-4,
[data-glamor-4] {
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 16px;
  -ms-grid-template-columns: 1fr;
  -ms-grid-gap: 16px;
}

.glamor-2,
[data-glamor-2] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #999;
}

.glamor-3,
[data-glamor-3] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
}

@media (min-width: 31em) {
  .glamor-5,
  [data-glamor-5] {
    width: 496px;
  }
}

@media (min-width: 31em) {
  .glamor-1,
  [data-glamor-1] {
    margin-bottom: 64px;
  }
}

@media (min-width: 31em) {
  .glamor-4,
  [data-glamor-4] {
    grid-template-columns: 1fr 1fr;
    -ms-grid-template-columns: 1fr 1fr;
  }
}

<div
  className="glamor-5"
>
  <h2
    className="glamor-0"
  >
    Are you sure?
  </h2>
  <div
    className="glamor-1"
  >
    Clicking yes will confirm the action. Clicking cancel will abort this dialog.
  </div>
  <div
    className="glamor-4"
  >
    <button
      className="glamor-2"
      onClick={[Function]}
      role="button"
    >
      cancel
    </button>
    <button
      className="glamor-3"
      onClick={[Function]}
      role="button"
    >
      Yes, I'm sure
    </button>
  </div>
</div>
`;
